### PR TITLE
Greene nightly multi-agent launcher + date-scoped wandb groups

### PR DESCRIPTION
**Pull request by eugenevinitsky**
## Summary

WIP — adds a Greene nightly launcher for the multi-agent (oignons2-derived) config, mirroring the existing single-agent launcher, plus date-scoped wandb grouping for both.

- \`scripts/cluster_configs/nightly_best.yaml\` — multi-agent config (720k agents, 8 CARLA towns, 10B steps, gigaflow encoder + split network, perturbations on).
- \`scripts/launch_nightly_best.sh\` — Greene launcher; mirrors \`launch_single_agent.sh\` but defaults \`TIME=1800\` / \`MEM=192gb\` to fit the heavier multi-agent profile.
- Both launchers now append \`wandb_group=\${DATE_STAMP}\` to \`--args\` so a night's 3 seeds cluster under one date-named wandb group.
- Yamls now point at split wandb projects: \`single_agent_speed_run.yaml -> nightly-single\`, \`nightly_best.yaml -> nightly-multi\`.

### CODE REVIEW

File: scripts/cluster_configs/nightly_best.yaml [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+policy.critic_num_layers: 0
+policy.split_network: true
+policy.encoder_gigaflow: true
+policy.dropout: 0.0
+
+# Training — 10B steps, large minibatch, compiled bfloat16
+train.total_timesteps: 10_000_000_000
+train.learning_rate: 0.0005
+train.minibatch_size: 153600
+train.max_minibatch_size: 153600
```
Comment (vcharraut) [inline scripts/cluster_configs/nightly_best.yaml]: imo the bath size can be simplified to power of 2, with adv filtering and masking you never get perfect bs anyway

File: scripts/cluster_configs/nightly_best.yaml [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+policy.input_size: 256
+policy.backbone_hidden_size: 1024
+policy.backbone_num_layers: 3
+policy.actor_hidden_size: 1024
+policy.actor_num_layers: 0
+policy.critic_hidden_size: 1024
+policy.critic_num_layers: 0
+policy.split_network: true
+policy.encoder_gigaflow: true
+policy.dropout: 0.0
```
Comment (vcharraut) [inline scripts/cluster_configs/nightly_best.yaml]: this should be deprecated, especially the `policy.encoder_gigaflow`

File: scripts/cluster_configs/nightly_best.yaml [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+# Goal setup — three sequential waypoints, route-based placement [20, 60m]
+env.num_target_waypoints: 3
+env.min_waypoint_spacing: 20.0
+env.max_waypoint_spacing: 60.0
+env.goal_radius: 2.0
+env.goal_speed: 3.0
+
+# Observation shaping
+env.obs_slots_lane_n: 80
+env.obs_slots_boundary_n: 80
```
Comment (vcharraut) [inline scripts/cluster_configs/nightly_best.yaml]: I think the `env.obs_slots_boundary_n: 80` can be reduced, since it is not similar to gigaflow implen

File: scripts/cluster_configs/nightly_best.yaml [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+# randomization on), partner-blindness / phantom-braking perturbations
+# enabled. Keys here override pufferlib/config/ocean/drive.ini.
+#
+# Launch via scripts/launch_nightly_best.sh (3 seeds, date-stamped).
+
+# Environment — multi-agent gigaflow over all 8 local CARLA towns
+env.simulation_mode: gigaflow
+env.map_dir: pufferlib/resources/drive/binaries/carla
+env.num_maps: 8
+env.num_agents: 720000
```
Comment (vcharraut) [inline scripts/cluster_configs/nightly_best.yaml]: this will break the memory i think?

File: scripts/cluster_configs/nightly_best.yaml (hunk 1 above)
Comment (eugenevinitsky) [reply scripts/cluster_configs/nightly_best.yaml]: So this is not the values you use? We're just trying to match the best run as closely as we can here

File: scripts/cluster_configs/nightly_best.yaml (hunk 3 above)
Comment (eugenevinitsky) [reply scripts/cluster_configs/nightly_best.yaml]: Sure! What value do you use in your best runs?

File: scripts/cluster_configs/nightly_best.yaml (hunk 2 above)
Comment (eugenevinitsky) [reply scripts/cluster_configs/nightly_best.yaml]: Fixed in a9b5e29b (with the dead-key removal in b4998507): `policy.input_size`, `split_network`, `encoder_gigaflow`, and `dropout` no longer exist in the [policy] schema and each crashed the launch at argparse time. `split_network: true` is translated to its successor `shared_network: false`; the other three have no equivalent and are dropped.

_Generated by AI agent_

File: scripts/cluster_configs/nightly_best.yaml (hunk 4 above)
Comment (eugenevinitsky) [reply scripts/cluster_configs/nightly_best.yaml]: You're right — the obs rollout buffer is num_agents × bptt(128) × obs_dim(828) float32 on the GPU, so 720k agents ≈ 305 GB against the launcher's single 141 GB H200 (bfloat16 is AMP-only and doesn't shrink it). Fixed in a9b5e29b: reduced to 153600 (≈65 GB, and it matches minibatch_size so the auto batch splits into exactly 128 minibatches).

_Generated by AI agent_

File: scripts/cluster_configs/nightly_best.yaml (hunk 3 above)
Comment (eugenevinitsky) [reply scripts/cluster_configs/nightly_best.yaml]: Heads-up: the yaml is now trimmed to only deltas from drive.ini (4b09564c), and `obs_slots_boundary_n: 80` matched the ini default, so the line is gone from this file. If it should be reduced, that's now either an explicit override here or a drive.ini default change — still curious what value your best runs use.

_Generated by AI agent_

File: scripts/cluster_configs/nightly_best.yaml (hunk 4 above)
Comment (eugenevinitsky) [reply scripts/cluster_configs/nightly_best.yaml]: Update: the override is dropped entirely in d59d532b — the config now inherits the drive.ini default num_agents rather than pinning a value here.

_Generated by AI agent_
